flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecLookupJoin.java

```diff
@@ -541,16 +632,9 @@ private ProcessFunction<RowData, RowData> createSyncLookupJoinFunction(
                         isObjectReuseEnabled);
 
         Optional<RelDataType> temporalTableOutputType =
-                projectionOnTemporalTable != null
-                        ? Optional.of(
-                                RexUtil.createStructType(
-                                        unwrapTypeFactory(relBuilder), projectionOnTemporalTable))
-                        : Optional.empty();
-        RowType rightRowType =
-                projectionOnTemporalTable != null
-                        ? (RowType) toLogicalType(temporalTableOutputType.get())
-                        : tableSourceRowType;
-        GeneratedCollector<TableFunctionCollector<RowData>> generatedCollector =
+                getProjectionRowTypeOnTemporalTable(relBuilder);
```

Comment:
we can call  the `getProjectionRowTypeOnTemporalTable`  method in `if (projectionOnTemporalTable != null) ` branch, and the implementation of `getProjectionRowTypeOnTemporalTable` method can be simplified

There is some duplicated code in `createAsyncLookupJoin` method

Reply:
done